src/openfermion/utils/_bch_expansion.py

```diff
@@ -0,0 +1,188 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Module to efficiently compute the Baker-Campbell-Hausdorff formula."""
+
+import itertools
+from scipy.misc import comb, factorial
+
+def bch_expand(x, y, order=6):
+    """Compute log[e^x e^y] using the Baker-Campbell-Hausdorff formula
+    Args:
+        x: An operator for which multiplication and addition are supported.
+            For instance, a QubitOperator, FermionOperator or scipy sparse
+            matrix.
+        y: The same type as x.
+        order(int): The max degree of monomial with respect to X and Y
+            to truncate the BCH expansions.
+
+    Returns:
+        z: The truncated BCH operator.
+
+    Raises:
+        ValueError: operator x is not same type as operator y.
+        ValueError: invalid order parameter.
+        ValueError: order exceeds maximum order supported.
+    """
+    MAX_ORDER = 15
```

Comment:
Sorry but I really don't think this MAX_ORDER should be there at all. If you really want to have it you can have the code raise a warning but do not have it raise an exception, as it is doing below.